core/network/adapters/protobuf_state_response.hpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_ADAPTERS_PROTOBUF_STATE_RESPONSE
+#define KAGOME_ADAPTERS_PROTOBUF_STATE_RESPONSE
+
+#include "network/adapters/protobuf.hpp"
+
+#include "network/types/state_response.hpp"
+#include "scale/scale.hpp"
+
+namespace kagome::network {
+
+  template <>
+  struct ProtobufMessageAdapter<StateResponse> {
+    static size_t size(const StateResponse &t) {
+      return 0;
+    }
+
+    static std::vector<uint8_t>::iterator write(
+        const StateResponse &t,
+        std::vector<uint8_t> &out,
+        std::vector<uint8_t>::iterator loaded) {
+      ::api::v1::StateResponse msg;
+      for (const auto &entries : t.entries) {
+        auto *dst_entries = msg.add_entries();
+        dst_entries->set_state_root(entries.state_root.toString());
+        for (const auto &entry : entries.entries) {
+          auto *dst_entry = dst_entries->add_entries();
+          dst_entry->set_key(entry.key.toString());
+          dst_entry->set_value(entry.value.toString());
+        }
+        dst_entries->set_complete(entries.complete);
+      }
+      msg.set_proof(t.proof.toString());
+
+      const size_t distance_was = std::distance(out.begin(), loaded);
+      const size_t was_size = out.size();
+
+      out.resize(was_size + msg.ByteSizeLong());
+      msg.SerializeToArray(&out[was_size], msg.ByteSizeLong());
+
+      auto res_it = out.begin();
+      std::advance(res_it, std::min(distance_was, was_size));
```

Comment:
Doublicated several times. Can we move it somewhere in common code?

Reply:
What exactly do you mean?

Reply:
I saw such a code twice in request and response both.

Reply:
Done